### PR TITLE
Testing page for the App framework

### PR DESCRIPTION
**Pull request by javaguirre**
#### Summary

Describing how to use e2e tests when making changes to the app framework.

Screenshot 2023-01-20 at 11 31 02

### CODE REVIEW

**Comment by javaguirre** [comment]
sorry, I'm missing something, I need to update the docs, I'll let you know when it's ready, @mickmister if you know how to run this tests better let me know, thank you!

File: site/content/integrate/apps/testing/_index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+heading: "Apps testing"
+description: "How to test the App framework"
+weight: 81
+---
+
+This page describes how to test changes made to the App Framework.
+
+## E2E Testing
+
+To run end-2-end testing with cypress you need to do the following:
```
Comment (justinegeffen) [inline site/content/integrate/apps/testing/_index.md:12]:
```suggestion
To run end-to-end testing with Cypress you need to do the following:
```

File: site/content/integrate/apps/testing/_index.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+- Clone the [Mattermost app examples](https://github.com/mattermost/mattermost-app-examples) repository
+- Launch the mattermost server
```
Comment (justinegeffen) [inline site/content/integrate/apps/testing/_index.md:15]:
```suggestion
- Launch the Mattermost server
```

File: site/content/integrate/apps/testing/_index.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+```bash
+cd mattermost-app-examples
+docker compose up
+```
+
+Once you have it up and running:
+
+- Go to the cloned [Mattermost Apps](https://github.com/mattermost/mattermost-plugin-apps) repository
+- Then run the e2e testing
```
Comment (justinegeffen) [inline site/content/integrate/apps/testing/_index.md:25]:
```suggestion
- Then run the E2E testing
```

**Review by justinegeffen — COMMENTED** [review]
Thank you! Some minor adjustments but otherwise looks good to me! :)

**Review by azigler — COMMENTED** [review]
Really great PR @javaguirre! If I understand correctly, you need more time to finish the content, so feel free to ping me for a later review. I went ahead and reorganized your starting structure for clarity, based on my own experiences copyediting this style of tutorial. I hope those edits will help you build a strong resource for the docs. Feel free to change however is best for you! 💪

**Comment by javaguirre** [comment]
> Really great PR @javaguirre! If I understand correctly, you need more time to finish the content, so feel free to ping me for a later review. I went ahead and reorganized your starting structure for clarity, based on my own experiences copyediting this style of tutorial. I hope those edits will help you build a strong resource for the docs. Feel free to change however is best for you! 💪

You're super helpful @azigler ! I'll do that, thank you for the changes!

**Comment by javaguirre** [comment]
> Really great PR @javaguirre! If I understand correctly, you need more time to finish the content, so feel free to ping me for a later review. I went ahead and reorganized your starting structure for clarity, based on my own experiences copyediting this style of tutorial. I hope those edits will help you build a strong resource for the docs. Feel free to change however is best for you! 💪

You're super helpful @azigler ! I'll do that, thank you for the changes!

**Comment by javaguirre** [comment]
I have to complte with [this](https://github.com/mattermost/mattermost-plugin-apps#end-to-end-tests) and other testing related tools (gomock, unittesting)

File: site/content/integrate/apps/testing/_index.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+```bash
+git clone https://github.com/mattermost/mattermost-app-examples
+cd mattermost-app-examples
+```
+
+2. Launch the provided Mattermost server
+
+```bash
+docker compose up
```
Comment (mickmister) [inline site/content/integrate/apps/testing/_index.md:24]: A little unfortunate that the e2e tests rely on the separate repo. Maybe have a small excerpt explaining why we need to clone this repo? Something like

> We will use this project's Docker Compose configuration to run our test server.

**Review by mickmister — APPROVED** [review]

File: site/content/integrate/apps/testing/_index.md (hunk 4 above)
Comment (javaguirre) [reply site/content/integrate/apps/testing/_index.md:24]: it's true, I'm not sure if it's the case or not, but, I'll continue investigating, thank you!

**Pull request closed without merging**